docs/shared/guides/unknown-local-cache.md

```diff
@@ -0,0 +1,72 @@
+
+```
+NX Invalid Cache Directory for Task "myapp:build"
+
+The local cache artifact in "node_modules/.cache/nx/786524780459028195" has not been generated on this machine.
+As a result, the cache's content integrity cannot be confirmed, which may make cache restoration potentially unsafe.
+If your machine ID has changed since the artifact was cached, run "nx reset" to fix this issue.
+Read about the error and how to address it here: https://nx.dev/recipes/other/unknown-local-cache
+```
+
+## Nx Tracks Cache Source
+
+Nx can cache tasks, which can drastically speed up your CI and local builds. However, this comes with the potential risk
+of "cache poisoning", where cache artifacts could be intentionally or inadvertently overwritten. If another user
+executes a task that matches the hash of the tainted artifact, they could retrieve the corrupted artifact. Nx and Nx
+Cloud contain several safeguards to minimize the likelihood of cache poisoning or, in the case of Nx Cloud, completely
+prevent it.
+
+The error above is one such safeguard.
+
+Nx trusts the local cache. If you executed a task and stored the corresponding cached artifact on your machine, you can
+safely restore it on the same machine without worrying about cache poisoning. The "Invalid Cache Directory" error is
+thrown when the local cache artifact was created on another machine.
+
+## You MachineId Changed
+
+Upgrading your computer's hardware may alter its machineId, yielding the error above. To fix it execute "nx reset" to
+remove all "invalid" cache directories. After doing so, you should no longer see the error.
+
+## You Share Cache with Another Machine Using a Network Drive
+
+You can prefix any Nx command with `NX_REJECT_UNKNOWN_LOCAL_CACHE=0` to ignore the error (
+e.g., `NX REJECT_UNKNOWN_LOCAL_CACHE=0 nx run-many -t build test`). This is similar to
+setting `NODE_TLS_REJECT_UNAUTHORIZED=0` to ignore any errors stemming form self-signed certificates. Even though it
+will make it work, this approach is discouraged.
+
+Sharing a network drive can present security risks. When a network drive is shared, every CI run has access to all the
+previously created Nx cache artifacts. Hence, it is plausible for every single artifact - for every single task hash -
+to be accessed without leaving any trace. This is feasible due to the network drive's capability to allow overwrites.
+
+Instead of sharing the network drive, we highly recommend you to implement the `RemoteCache` interface.
+
+## Implementing Remote Cache Interface
+
+This is the interface:
+
+```typescript
+interface RemoteCache {
+  retrieve(hash: string, cachePath: string);
+
+  store(hash: string, cachePath: string);
+}
+```
+
+You will need to wrap the default tasks runner to provide the remote cache implementation.
+
+## How Nx Cloud Makes Sure Sharing Cache is Safe
+
+The Nx Cloud runner provides an implementation of `RemoteCache` which does the following things:
```

Comment:
If having multiple machines write to the same cache is unsafe, you may now have some reasonable suspicion that Nx Cloud is unsafe as well. However, Nx Cloud has several safeguards to prevent cache poisoning.